src/backend/metal/src/command.rs

```diff
@@ -1191,24 +1218,101 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
             .blit_commands(iter::once(command));
     }
 
-    fn clear_color_image_raw(
+    fn clear_image<T>(
         &mut self,
-        _image: &native::Image,
+        image: &native::Image,
         _layout: Layout,
-        _range: SubresourceRange,
-        _value: com::ClearColorRaw,
-    ) {
-        unimplemented!()
-    }
+        color: com::ClearColorRaw,
+        depth_stencil: com::ClearDepthStencilRaw,
+        subresource_ranges: T,
+    ) where
+        T: IntoIterator,
+        T::Item: Borrow<SubresourceRange>,
+    {
+        let mut inner = self.inner.borrow_mut();
+        inner.sink().stop_encoding();
 
-    fn clear_depth_stencil_image_raw(
-        &mut self,
-        _image: &native::Image,
-        _layout: Layout,
-        _range: SubresourceRange,
-        _value: com::ClearDepthStencilRaw,
-    ) {
-        unimplemented!()
+        let clear_color = unsafe {
+            match image.shader_channel {
+                Channel::Float =>metal::MTLClearColor::new(
+                    color.float32[0] as _,
+                    color.float32[1] as _,
+                    color.float32[2] as _,
+                    color.float32[3] as _,
+                ),
+                Channel::Int => metal::MTLClearColor::new(
+                    color.int32[0] as _,
+                    color.int32[1] as _,
+                    color.int32[2] as _,
+                    color.int32[3] as _,
+                ),
+                Channel::Uint => metal::MTLClearColor::new(
+                    color.uint32[0] as _,
+                    color.uint32[1] as _,
+                    color.uint32[2] as _,
+                    color.uint32[3] as _,
+                ),
+            }
+        };
+
+        for subresource_range in subresource_ranges {
+            let sub = subresource_range.borrow();
+            let end_level = if sub.levels.end == !0 {
+                image.raw.mipmap_level_count() as _
+            } else {
+                sub.levels.end
+            };
+            let end_layer = if sub.layers.end == !0 {
+                image.raw.array_length() as _
+            } else {
+                sub.layers.end
+            };
+
+            for level in sub.levels.start .. end_level {
+                for layer in sub.layers.start .. end_layer {
+                    let descriptor = metal::RenderPassDescriptor::new().to_owned();
```

Comment:
IMO we may want to remove this function from metal-rs – it seems a bit sketchy because it creates [`MTLRenderPassDescriptorInternal`](https://github.com/gfx-rs/metal-rs/blob/master/src/renderpass.rs#L292) directly. There is a [`renderPassDescriptor`](https://developer.apple.com/documentation/metal/mtlrenderpassdescriptor/1437979-renderpassdescriptor) convenience function that should set up the default attachments that we use here though.

Reply:
good point! filed https://github.com/gfx-rs/metal-rs/issues/50